apps/openmw/mwmechanics/aicombataction.cpp

```diff
@@ -594,13 +592,17 @@ namespace MWMechanics
                     return 0.f;
             }
         }
-        else
-        {
-            rating *= (effect.mMagnMin + effect.mMagnMax)/2.f;
-        }
 
-        if (!(magicEffect->mData.mFlags & ESM::MagicEffect::NoDuration))
-            rating *= effect.mDuration;
+        rating *= ((effect.mMagnMin + effect.mMagnMax) * (effect.mDuration > 0 ? effect.mDuration : 1) + effect.mArea);
```

Comment:
Why should we use magnMin / magnMax for a spell with NoMagnitude? Same question for duration, too. I wouldn't expect these variables to have a sane default value if the respective 'don't use' flag is set.